Make fundamentals workshop usable without a Moderne license (#846)
moderneinc/moderne-docs: docs/hands-on-learning/fundamentals/module-1-recipe-development-environment.md

#### Step 1: Clone and open the project

1. Clone or fork the [`rewrite-recipe-starter`](https://github.com/moderneinc/rewrite-recipe-starter) project or [use it as a template](https://github.com/new?template_name=rewrite-recipe-starter&template_owner=moderneinc).
2. Open the project in your IDE (this workshop demonstrates IntelliJ IDEA).

> 3. If you're using IntelliJ, choose Maven or Gradle import when prompted (your preference), and enable annotation processing if the IDE suggests it. These steps will differ if you're using the command line or another IDE.

Reviewer: ```suggestion
3. Import either the Maven or Gradle build (your preference) into your IDE and enable annotation processing. 
```

If they're using an IDE that does not delegate to a build tool as IntelliJ does by default, they might need to manually edit their IDE's java compiler options to include `-parameters`. Without it the test framework might report that their recipe fails a serialization check. Failing that serialization check would mean that the CLI would not be able to load their recipe later.

Author: Added a note about this.

mod config recipes jar install com.yourorg:rewrite-recipe-starter:LATEST
```

3. To confirm that everything is set up for testing imperative recipes, run one of the starter project's recipes by name. Open a terminal, navigate to your workshop directory, and run:

  ```bash
  cd ~/moderne-workshop

>   mod run . --recipe=com.yourorg.AssertEqualsToAssertThat

Reviewer: Without the IntelliJ plugin handling in there are a few manual steps they have to follow.
They can either build and publish the full recipe jar each time they want to try it out:
```
# Either of these will place the reicpe module jar & pom into ~/.m2/repository
# Must be run *every time* the developer changes the in-development recipe
./mvnw clean install
./gradlew publishToMavenLocal

# This only needs to be run once
mod config recipes jar install <coordinates of rewrite-recipe-starter>

mod run <test set dir> --recipe=...
```

OR they can do what the IntelliJ plugin does under the hood:
```
# This only needs to be run once
mod config recipes active set src/main/java/com/example/MyRecipe.java                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
mod config recipes active set src/main/resources/META-INF/rewrite/my-recipe.yml                                                                                                                                                                  
                                                                                                                                                                                                                                                   
# Must be run *every time* developer changes the in-development recipe
# Optionally add --continuous to the gradle invocation and gradle will watch the file system for changes and automatically re-run the build for every change
./gradlew classes
./mvnw compile
                                                                                                                                                                                                                      
# Run :                                                                                                                                                                                                                     
mod run <test set dir> --active-recipe                                                                                                                                                                                                                        
```

Author: Thanks Sam for this clarification. Updated accordingly. TIL `recipes active set` command.
